src/core/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,337 @@
+          event.preventDefault();
+          event.stopPropagation();
+          break;
+        case 'ArrowUp':
+          if (isOpen) {
+            const previousIndex = Math.max(focusedIndexInFilteredList - 1, 0);
+            setFocusedIndex(options.indexOf(filteredOptions[previousIndex]));
+          }
+          event.preventDefault();
+          event.stopPropagation();
+          break;
+        case 'Enter':
+          isOpen && setSelectedValue(options[focusedIndex].value);
+          setIsOpen(!isOpen);
+          event.preventDefault();
+          event.stopPropagation();
+          break;
+        case 'Escape':
+          setIsOpen(false);
+          event.preventDefault();
+          event.stopPropagation();
+          break;
+        case 'Tab':
+          setIsOpen(false);
+          break;
+        default:
+          !isOpen && setIsOpen(true); // reopen menu if closed when typing
+          break;
+      }
+    },
+    [filteredOptions, focusedIndex, isOpen, options],
+  );
+
+  const menuItems = React.useCallback(
+    (close: () => void) => {
+      if (filteredOptions.length === 0) {
+        return <MenuItem disabled>{emptyStateMessage}</MenuItem>;
+      }
+      return filteredOptions.map((option) => {
+        const index = options.findIndex(({ value }) => option.value === value);
+        const { label, value, ...rest } = option;
+        return (
+          <MenuItem
+            id={getOptionId(index)}
+            key={getOptionId(index)}
+            value={value}
+            className={cx({ 'iui-focused': focusedIndex === index })}
+            onClick={(value: T) => {
+              setSelectedValue(value);
+              close();
+            }}
+            isSelected={selectedValue === value}
+            ref={(el) => {
+              if (
+                focusedIndex === index ||
+                (focusedIndex === -1 && selectedValue === value)
+              ) {
+                el?.scrollIntoView(false);
+              }
+            }}
+            role='option'
+            {...rest}
+          >
+            {label}
+          </MenuItem>
+        );
+      });
+    },
+    [
+      filteredOptions,
+      emptyStateMessage,
+      options,
+      getOptionId,
+      focusedIndex,
+      selectedValue,
+    ],
+  );
+
+  return (
+    <InputContainer
+      className={className}
+      isIconInline={true}
+      icon={
+        isOpen ? (
+          <SvgCaretUpSmall
+            style={{ cursor: 'pointer' }}
+            onClick={() => setIsOpen(false)}
+          />
+        ) : (
+          <SvgCaretDownSmall
+            style={{ cursor: 'pointer' }}
+            onClick={() => inputRef.current?.focus()}
+          />
+        )
+      }
+      {...rest}
+      id={id}
+    >
+      <Popover
+        placement='bottom-start'
+        visible={isOpen}
+        onClickOutside={() => setIsOpen(false)}
+        {...dropdownMenuProps}
+        content={
+          <Menu
+            id={`${id}-list`}
```

Comment:
Maybe move `id` to the end, looks weird having it at the beginning.

Reply:
if the id was "my-combobox", this would be "my-combobox-list" which makes perfect sense to me (and "list-my-combobox" doesn't)

Reply:
I imagine id to be `1234-asd-789` and for me it looks weird `123-asd-qwe-list` 😀

Reply:
um, id cannot start with a number. So it would be `iui-1234-asd-qwe-list`.

If id is provided by user, it will likely be something more useful too. `country-select-list`

Reply:
It really doesn't change the fact if it starts with `123` or `asd`.

> If id is provided by user, it will likely be something more useful too.

That's why I said to make it mandatory, I don't really see a big problem here.